core/src/main/scala/sttp/tapir/server/interpreter/DecodeBasicInputs.scala

```diff
@@ -81,8 +81,8 @@ object DecodeBasicInputs {
     // we're using null as a placeholder for the future values. All except the body (which is determined by
     // interpreter-specific code), should be filled by the end of this method.
     compose(
-      matchOthers(methodInputs, _, _),
       matchPath(pathInputs, _, _),
+      matchOthers(methodInputs, _, _),
```

Comment:
In the case where both the path and method don't match, 404 should be returned, so we want the path match to fail first and be propagated out.